### ISSUE TITLE
Database connection in DatabaseSessionHandler is not set correctly.

### ISSUE THREAD

**Issue by crlcu** [archtechx/tenancy#601]
#### Describe the bug
The sessions table needs to be on the central database when using Jetstream

#### Steps to reproduce
Once Jetstream is installed, comment out the
// require __DIR__.'/auth.php';
from web.php in order to make use of the routes defined in Jetstream/Fortify (I've added the middleware in jetstream.php and fortify.php)

#### Expected behavior
The database connection should have the credentials from tenant instead of central domain
Dump from `Illuminate\Session\DatabaseSessionHandler`, line 64:
![image](https://user-images.githubusercontent.com/1661844/108165838-e79f2180-70fb-11eb-8585-c6f3ffd6f88a.png)
Dump from `Stancl\Tenancy\Middleware\IdentificationMiddleware`, line 36:
![image](https://user-images.githubusercontent.com/1661844/108165925-13baa280-70fc-11eb-8e45-b98524a7eb69.png)

#### Your setup
 - Laravel version: 8.12
 - stancl/tenancy version: 3.4

**Comment by stancl** [comment]
> The sessions table needs to be on the central database when using Jetstream

Hmm why exactly?

**Comment by X-Coder** [comment]
For me the issue is the other way around:
Jetstream is using the session table of central database instead of the tenant's one for authentication, when it is in tenant context.
I think it should use tenant's session table for this purpose.

I traced it back into laravel\framework\src\Illuminate\Session\SessionManager.php:
```php
    protected function getDatabaseConnection()
    {
        $connection = $this->config->get('session.connection'); // <- uses central instead of tenant, when in tenant context
        return $this->container->make('db')->connection($connection);
    }
```

```php
// Traceback
Illuminate\Session\SessionManager->getDatabaseConnection (vendor\laravel\framework\src\Illuminate\Session\SessionManager.php:101)
Illuminate\Session\SessionManager->createDatabaseDriver (vendor\laravel\framework\src\Illuminate\Session\SessionManager.php:90)
Illuminate\Session\SessionManager->createDriver (vendor\laravel\framework\src\Illuminate\Support\Manager.php:105)
Illuminate\Session\SessionManager->driver (vendor\laravel\framework\src\Illuminate\Support\Manager.php:80)
Illuminate\Session\SessionServiceProvider->Illuminate\Session\{closure:vendor\laravel\framework\src\Illuminate\Session\SessionServiceProvider.php:48-53} (vendor\laravel\framework\src\Illuminate\Session\SessionServiceProvider.php:52)
Illuminate\Foundation\Application->build (vendor\laravel\framework\src\Illuminate\Container\Container.php:826)
Illuminate\Foundation\Application->resolve (vendor\laravel\framework\src\Illuminate\Container\Container.php:712)
Illuminate\Foundation\Application->resolve (vendor\laravel\framework\src\Illuminate\Foundation\Application.php:841)
Illuminate\Foundation\Application->make (vendor\laravel\framework\src\Illuminate\Container\Container.php:651)
Illuminate\Foundation\Application->make (vendor\laravel\framework\src\Illuminate\Foundation\Application.php:826)
Illuminate\Foundation\Application->offsetGet (vendor\laravel\framework\src\Illuminate\Container\Container.php:1358)
Illuminate\Routing\RoutingServiceProvider->Illuminate\Routing\{closure:vendor\laravel\framework\src\Illuminate\Routing\RoutingServiceProvider.php:111-122} (vendor\laravel\framework\src\Illuminate\Routing\RoutingServiceProvider.php:118)
Illuminate\Foundation\Application->build (vendor\laravel\framework\src\Illuminate\Container\Container.php:826)
Illuminate\Foundation\Application->resolve (vendor\laravel\framework\src\Illuminate\Container\Container.php:712)
Illuminate\Foundation\Application->resolve (vendor\laravel\framework\src\Illuminate\Foundation\Application.php:841)
Illuminate\Foundation\Application->make (vendor\laravel\framework\src\Illuminate\Container\Container.php:651)
Illuminate\Foundation\Application->make (vendor\laravel\framework\src\Illuminate\Foundation\Application.php:826)
Illuminate\Foundation\Application->offsetGet (vendor\laravel\framework\src\Illuminate\Container\Container.php:1358)
Illuminate\Routing\RoutingServiceProvider->Illuminate\Routing\{closure:vendor\laravel\framework\src\Illuminate\Routing\RoutingServiceProvider.php:167-169} (vendor\laravel\framework\src\Illuminate\Routing\RoutingServiceProvider.php:168)
Illuminate\Foundation\Application->build (vendor\laravel\framework\src\Illuminate\Container\Container.php:826)
Illuminate\Foundation\Application->resolve (vendor\laravel\framework\src\Illuminate\Container\Container.php:712)
Illuminate\Foundation\Application->resolve (vendor\laravel\framework\src\Illuminate\Foundation\Application.php:841)
Illuminate\Foundation\Application->make (vendor\laravel\framework\src\Illuminate\Container\Container.php:651)
Illuminate\Foundation\Application->make (vendor\laravel\framework\src\Illuminate\Foundation\Application.php:826)
app (vendor\laravel\framework\src\Illuminate\Foundation\helpers.php:119)
response (vendor\laravel\framework\src\Illuminate\Foundation\helpers.php:687)
App\Exceptions\Handler->renderHttpException (vendor\laravel\framework\src\Illuminate\Foundation\Exceptions\Handler.php:553)
App\Exceptions\Handler->prepareResponse (vendor\laravel\framework\src\Illuminate\Foundation\Exceptions\Handler.php:461)
App\Exceptions\Handler->render (vendor\laravel\framework\src\Illuminate\Foundation\Exceptions\Handler.php:339)
Illuminate\Routing\Pipeline->handleException (vendor\laravel\framework\src\Illuminate\Routing\Pipeline.php:51)
Illuminate\Routing\Pipeline->Illuminate\Pipeline\{closure:vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:126-132} (vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:130)
Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull->handle (vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\TransformsRequest.php:21)
Illuminate\Routing\Pipeline->Illuminate\Pipeline\{closure:vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:143-174} (vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:167)
App\Http\Middleware\TrimStrings->handle (vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\TransformsRequest.php:21)
Illuminate\Routing\Pipeline->Illuminate\Pipeline\{closure:vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:143-174} (vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:167)
Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle (vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\ValidatePostSize.php:27)
Illuminate\Routing\Pipeline->Illuminate\Pipeline\{closure:vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:143-174} (vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:167)
App\Http\Middleware\PreventRequestsDuringMaintenance->handle (vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance.php:86)
Illuminate\Routing\Pipeline->Illuminate\Pipeline\{closure:vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:143-174} (vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:167)
Fruitcake\Cors\HandleCors->handle (vendor\fruitcake\laravel-cors\src\HandleCors.php:37)
Illuminate\Routing\Pipeline->Illuminate\Pipeline\{closure:vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:143-174} (vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:167)
App\Http\Middleware\TrustProxies->handle (vendor\fideloper\proxy\src\TrustProxies.php:57)
Illuminate\Routing\Pipeline->Illuminate\Pipeline\{closure:vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:143-174} (vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:167)
Illuminate\Routing\Pipeline->then (vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php:103)
App\Http\Kernel->sendRequestThroughRouter (vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php:141)
App\Http\Kernel->handle (vendor\laravel\framework\src\Illuminate\Foundation\Http\Kernel.php:110)
{main} (public\index.php:52)
```
It seems tenancy is not initialized when this code runs.
I try to find a solution for this, maybe someone know one already?

**Comment by crlcu** [comment]
Well, its the same way for me as well.
Let me try to explain again: It tries to find the sessions table on the central database, no matter if I’m in the tenant context or not.

**Comment by X-Coder** [comment]
I think I got mine working now with database session driver, by following these steps:

- [Enable Universal routes](https://tenancyforlaravel.com/docs/v3/features/universal-routes)
  - add (as described) 'universal' to config of jetstream, fortify, livewire, and route.
  - skip adding 'InitializeTenancyByDomain' because it will be initialized globally by the next step
- apply Rory's fix: https://stackoverflow.com/a/66261097
    this changes middleware InitializeTenancyByDomain to be loaded globally except for central domains to fix a endless loop at login caused by a not separated session

**Comment by stancl** [comment]
Jetstream support is not an offered feature currently, there are some needed changes to the dependency injection in Jetstream etc. It was discussed on our Discord and I'll be implementing it in the new boilerplate.

Is there any bug specifically in the package? As in something that should be working a certain way per the docs but isn't?

If so, please open an issue for that. This has a lot of Jetstream-specific stuff that's hard to debug efficiently.
